Generals/Code/GameEngine/Include/GameLogic/Object.h

```diff
@@ -237,6 +237,8 @@ class Object : public Thing, public Snapshot
 	void setCustomIndicatorColor(Color c);
 	void removeCustomIndicatorColor();
 
+	Bool isLogicallyVisible() const; ///< Returns whether the object is logically visible to the player.
```

Comment:
I suggest also clarify that this does not consider shroud, which is an important factor.